v3/tags.go

```diff
@@ -253,10 +254,8 @@ func (ti *TagIndex) getOne(ifdPath string, id uint16) (it *IndexedTag, err error
 		}
 	}()
 
-	if len(ti.tagsByIfd) == 0 {
-		err := LoadStandardTags(ti)
-		log.PanicIf(err)
-	}
+	err = LoadStandardTags(ti)
```

Comment:
This is in subtle violation of the original intent. As originally written, it only loads the standard tags if the user hasn't already loaded anything. If the user is going to load tags, they could be the standard tags, provided for convenience, or a completely alternative set of tags. There may be obscure scenarios where the user wants to not have any of the standard-tag handling. With this guard removed, the user will now always have the standard tags loaded.

I recommend you restore the original design and move the `sync.Once()` call to here (passing `LoadStandardTags` as the argument). There also won't be any net difference in the behavior or user requirements. Alternatively, you could just wrap the guards and `LoadStandardTags()` in a lock/unlock-based mutex. All things being equal, I prefer those to `sync.Once`.

Reply:
I have pushed a commit that adds back the check for `tagsByIfd` inside the sync.Once. I don't see how to avoid data race by a simple mutex, but please prove me wrong, you know this code better than me. If I could suggest something  it would be to let the client pass in everything in the constructor; that way you don't need these locks.